include/oneapi/dpl/internal/random_impl/philox_engine.h

```diff
@@ -0,0 +1,483 @@
+// -*- C++ -*-
+//===-- philox-engine.h ------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Abstract:
+//
+// Public header file provides implementation for Philox Engine
+
+#ifndef _ONEDPL_PHILOX_ENGINE_H
+#define _ONEDPL_PHILOX_ENGINE_H
+
+#include "random_common.h"
+#include "counter_based_engines_stuff.h"
+
+namespace oneapi
+{
+namespace dpl
+{
+namespace experimental
+{
+
+template <typename _UIntType, ::std::size_t _w, ::std::size_t _n, ::std::size_t _r,
+          internal::element_type_t<_UIntType>... _consts>
+class philox_engine;
+
+template <class _CharT, class _Traits, typename __UIntType, ::std::size_t __w, ::std::size_t __n, ::std::size_t __r,
+          __UIntType... __consts>
+::std::basic_ostream<_CharT, _Traits>&
+operator<<(::std::basic_ostream<_CharT, _Traits>&, const philox_engine<__UIntType, __w, __n, __r, __consts...>&);
+
+template <typename __UIntType, ::std::size_t __w, ::std::size_t __n, ::std::size_t __r, __UIntType... __consts>
+const sycl::stream&
+operator<<(const sycl::stream&, const philox_engine<__UIntType, __w, __n, __r, __consts...>&);
+
+template <class _CharT, class _Traits, typename __UIntType, ::std::size_t __w, ::std::size_t __n, ::std::size_t __r,
+          __UIntType... __consts>
+::std::basic_istream<_CharT, _Traits>&
+operator>>(::std::basic_istream<_CharT, _Traits>&, philox_engine<__UIntType, __w, __n, __r, __consts...>&);
+
+template <typename _UIntType, ::std::size_t _w, ::std::size_t _n, ::std::size_t _r,
+          internal::element_type_t<_UIntType>... _consts>
+class philox_engine
+{
+    /* The size of the consts arrays */
+    static constexpr ::std::size_t array_size = _n / 2;
+
+  public:
+    /* Types */
+    using result_type = _UIntType;
+    using scalar_type = internal::element_type_t<result_type>;
+
+    /* Engine characteristics */
+    static constexpr ::std::size_t word_size = _w;
+    static constexpr ::std::size_t word_count = _n;
+    static constexpr ::std::size_t round_count = _r;
+
+    static_assert(_n == 2 || _n == 4, "_n must be 2 or 4");
+    static_assert(sizeof...(_consts) == _n, "the amount of _consts must be equal to _n");
+    static_assert(_r > 0, "_r must be more than 0");
+    static_assert(_w > 0 && _w <= ::std::numeric_limits<scalar_type>::digits,
+                  "_w must satisfy 0 < _w < ::std::numeric_limits<_UIntType>::digits");
+    static_assert(::std::numeric_limits<scalar_type>::digits <= 64,
+                  "size of the scalar _UIntType (in case of sycl::vec<T, N> the size of T) must be less than 64 bits");
+    static_assert(::std::is_unsigned_v<scalar_type>, "_UIntType must be unsigned type or vector of unsigned types");
+
+    static constexpr ::std::array<scalar_type, array_size> multipliers =
```

Comment:
```suggestion
    static constexpr std::array<scalar_type, array_size> multipliers =
```

In the new code oneDPL stops using :: before std namespace

Reply:
All occurrences were refactored